pkg/platform/runtime/buildscript/buildscript.go

```diff
@@ -266,14 +268,26 @@ func transformVersion(requirements *buildexpression.Var) *buildexpression.Ap {
 	}
 
 	if len(aps) == 1 {
-		return aps[0] // e.g. Eq("1.0")
+		return aps[0] // e.g. Eq(value = "1.0")
 	}
 
-	args := make([]*buildexpression.Value, len(aps))
-	for i, ap := range aps {
-		args[i] = &buildexpression.Value{Ap: ap}
+	// e.g. And(left = Gt(value = "1.0"), right = Lt(value = "3.0"))
+	// Iterate backwards over the requirements array and construct a binary tree of 'And()' functions.
+	// For example, given [Gt(value = "1.0"), Ne(value = "2.0"), Lt(value = "3.0")], produce:
+	//   And(left = Gt(value = "1.0"), right = And(left = Ne(value = "2.0"), right = Lt(value = "3.0")))
```

Comment:
@srobertson could you confirm this is what you want? If `And()` only takes 2 parameters, then this is all we can reasonably do. But I could also see an argument for making `And()` take 2+ parameters, in which case we'd also need to revisit the naming.

To summarize the comment, technically you can have a version constraint of:

```
>1.0,<3.0,!=2.0
```

Which I think ideally translates to something like:

```
And(value1 = Gt(value="1.0"), value2 = Lt("3.0"), value3 = Ne("3.0"))
```

But as we only have the left and right parameters, we instead end up with:

```
And(left = Gt(value="1.0"), right = And(left = Lt("3.0"), right = Ne("3.0"))
```